react_app/src/components/ChallengeWidget.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { makeStyles } from "@material-ui/core/styles";
+
+import { GameContext } from "../Context";
+
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Typography from "@material-ui/core/Typography";
+import Dialog from "@material-ui/core/Dialog";
+import Grid from "@material-ui/core/Grid";
+
+import FlagWidget from "./Flag";
+
+const styles = makeStyles(theme => ({
+  root: {
+    width: "60%",
+    height: "80%",
+    padding: theme.spacing()
+  }
+}));
+
+class ChallengeWidget extends React.Component {
+  static contextType = GameContext;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: true
+    };
+    this.handleClose = this.handleClose.bind(this)
+  }
+
+  handleClose() {
+    this.context.close()
+  }
+
+  render() {
+    const challenge = this.props.details;
+
+    return (
+      <div>
+        <Dialog onClose={this.handleClose} aria-labelledby="challenge-dialog" className={styles.root} open={true}>
```

Comment:
So this dialog will always be open if this component is visible?

Reply:
Yeah, this seemed a touch hacky, but the dialog is controlled by its ancestor's state - because I'm going to have possibly dozens of challengecards rendered at once, if I were to create dialoges for each of those but set them to closed you'd have a queue of get requests - this approach loads them as needed instead